rp2040-hal/src/timer.rs

```diff
@@ -199,6 +205,22 @@ pub trait Alarm {
         countdown: Duration<u32, NOM, DENOM>,
     ) -> Result<(), ScheduleAlarmError>;
 
+    /// Schedule the alarm to be finished at the given timestamp. If [enable_interrupt] is
+    /// called, this will trigger interrupt whenever this timestamp is reached.
+    ///
+    /// The RP2040 has been observed to take a little while to schedule an alarm. For this
+    /// reason, this function requires the timestamp to be at least 10 microseconds in the
+    /// future.
```

Comment:
This claim sounds a little bit dubious. Why should the rp2040 take microseconds to schedule an alarm?
Is there some example available which demonstrates this delay?

Reply:
This is the same claim as on `::schedule`. I did not witness that myself.
It'd rather see the sequence in a time critical block:

- set timestamp
- check if armed
- check if counter is > timestamp
- check if still armed
- if yes => we set the alarm in the past => error

Reply:
The new code looks much more sensible, thanks!
→ Now the comment is outdated :smile: